tests/std/tests/P2321R2_views_adjacent_transform/test.cpp

```diff
@@ -926,6 +925,60 @@ constexpr void instantiation_test() {
 #endif // TEST_EVERYTHING
 }
 
+// LWG-4098 views::adjacent<0> should reject non-forward ranges
+template <size_t N, ranges::input_range Rng, class Fn>
+constexpr void test_input_only(Rng&&, Fn&&) {
+    if constexpr (!ranges::forward_range<Rng>) {
+        static_assert(!CanViewAdjacentTransform<Rng&, Fn, N>);
+        static_assert(!CanViewAdjacentTransform<Rng, Fn, N>);
+    }
+
+    if constexpr (!ranges::forward_range<const Rng>) {
+        static_assert(!CanViewAdjacentTransform<const Rng&, Fn, N>);
+        static_assert(!CanViewAdjacentTransform<const Rng, Fn, N>);
+    }
+}
+
+struct input_only_instantiator {
+#ifdef TEST_EVERYTHING
+    template <ranges::input_range R>
+    static constexpr void call() {
+        R r{some_ints};
+        auto nullary_fn = [] { return 1729; };
+        test_input_only<0>(r, nullary_fn);
+        test_input_only<1>(r, adjacent1_fn);
+        test_input_only<2>(r, pairwise_fn);
+        test_input_only<4>(r, adjacent4_fn);
+        test_input_only<7>(r, adjacent7_fn);
+    }
+#else // ^^^ test all input range permutations / test only "interesting" permutations vvv
+    template <class Tag, test::Common IsCommon, test::Sized IsSized>
+    static constexpr void call() {
+        test_range<Tag, IsCommon, IsSized> r{some_ints};
+        auto nullary_fn = [] { return 1729; };
+        test_input_only<0>(r, nullary_fn);
+        test_input_only<1>(r, adjacent1_fn);
+        test_input_only<2>(r, pairwise_fn);
+        test_input_only<4>(r, adjacent4_fn);
+        test_input_only<7>(r, adjacent7_fn);
+    }
+#endif // TEST_EVERYTHING
+};
+
+constexpr void instantiation_input_only_test() {
+#ifdef TEST_EVERYTHING
+    test_in<input_only_instantiator, const int>();
+#else // ^^^ test all forward range permutations / test only "interesting" permutations vvv

```

Comment:
This comment also says "forward" when it should say "input". (The comment on 954 is already doing this.)